packages/tools/src/tools/annotation/UltrasoundAnnotationTool/utils/deriveFanGeometry.ts

```diff
@@ -0,0 +1,97 @@
+import type { Types } from '@cornerstonejs/core';
+import type { FanGeometry, FanShapeCorners } from './types';
+
+/**
+ * Compute intersection of two infinite lines (p1→p2) and (p3→p4).
+ * Returns a point or null if parallel.
+ *
+ * @param {Types.Point2} p1 - First point of first line
+ * @param {Types.Point2} p2 - Second point of first line
+ * @param {Types.Point2} p3 - First point of second line
+ * @param {Types.Point2} p4 - Second point of second line
+ * @returns {Types.Point2|null} Intersection point or null if lines are parallel
+ */
+function intersect(
```

Comment:
we already have this packages/tools/src/utilities/math/line/intersectLine.ts see if you can use it

Reply:
I´ve add an parameter infinite to check infinite lines intersection in the intersectLine, ok? Because both fan lateral lines do not intersect as line segments and intersectLine checks only intersections between line segments